Mathlib/NumberTheory/Harmonic/ZetaAsymp.lean

```diff
@@ -0,0 +1,373 @@
+    · exact continuousAt_const.rpow (continuousAt_id.add continuousAt_const) (Or.inr (by linarith))
+    · exact (rpow_pos_of_pos ((Nat.cast_pos.mpr (by simp)).trans hx.1) _).ne'
+
+lemma continuousOn_term_tsum : ContinuousOn term_tsum (Ici 1) := by
+  -- We use dominated convergence, using `fun n ↦ term n 1` as our uniform bound (since `term` is
+  -- monotone decreasing in `s`.)
+  refine continuousOn_tsum (fun i ↦ continuousOn_term _) term_tsum_one.summable (fun n s hs ↦ ?_)
+  rw [term, term, norm_of_nonneg]
+  · simp_rw [intervalIntegral.integral_of_le (by linarith : (↑(n + 1) : ℝ) ≤ ↑(n + 1) + 1)]
+    refine set_integral_mono_on ?_ ?_ measurableSet_Ioc (fun x hx ↦ ?_)
+    · exact (term_welldef n.succ_pos (zero_lt_one.trans_le hs)).1
+    · exact (term_welldef n.succ_pos zero_lt_one).1
+    · rw [div_le_div_left] -- leave side-goals to end and kill them all together
+      apply rpow_le_rpow_of_exponent_le
+      · exact (lt_of_le_of_lt (by simp) hx.1).le
+      · linarith [mem_Ici.mp hs]
+      · linarith [hx.1]
+      all_goals apply rpow_pos_of_pos ((Nat.cast_nonneg _).trans_lt hx.1)
+  · rw [intervalIntegral.integral_of_le (by linarith)]
+    refine set_integral_nonneg measurableSet_Ioc (fun x hx ↦ div_nonneg ?_ (rpow_nonneg ?_ _))
+    all_goals linarith [hx.1]
+
+/-- First version of the limit formula, with a limit over real numbers tending to 1 from above. -/
+lemma tendsto_riemannZeta_nhds_right : Tendsto (fun s : ℝ ↦ riemannZeta s - 1 / (s - 1))
+    (𝓝[>] 1) (𝓝 eulerMascheroniConstant) := by
+  suffices Tendsto (fun s : ℝ ↦ (∑' n : ℕ, 1 / (n + 1 : ℝ) ^ s) - 1 / (s - 1))
+    (𝓝[>] 1) (𝓝 eulerMascheroniConstant) by
+    apply ((Complex.continuous_ofReal.tendsto _).comp this).congr'
+    filter_upwards [self_mem_nhdsWithin] with s hs
+    simp only [Function.comp_apply, Complex.ofReal_sub, Complex.ofReal_div,
+      Complex.ofReal_one, sub_left_inj, Complex.ofReal_tsum]
+    rw [zeta_eq_tsum_one_div_nat_add_one_cpow (by simpa using hs)]
+    congr 1 with n
+    rw [Complex.ofReal_cpow (by positivity)]
+    norm_cast
+  suffices aux2 : Tendsto (fun s : ℝ ↦ (∑' n : ℕ, 1 / (n + 1 : ℝ) ^ s) - 1 / (s - 1))
+    (𝓝[>] 1) (𝓝 (1 - term_tsum 1)) by
+    have := term_tsum_one.tsum_eq
+    rw [← term_tsum, eq_sub_iff_add_eq, ← eq_sub_iff_add_eq'] at this
+    simpa only [this] using aux2
+  apply Tendsto.congr'
+  · filter_upwards [self_mem_nhdsWithin] with s hs using (zeta_limit_aux1 hs).symm
+  · apply tendsto_const_nhds.sub
+    rw [← one_mul (term_tsum 1)]
+    apply (tendsto_id.mono_left nhdsWithin_le_nhds).mul
+    have := continuousOn_term_tsum.continuousWithinAt left_mem_Ici
+    exact Tendsto.mono_left this (nhdsWithin_mono _ Ioi_subset_Ici_self)
+
+/-- The function `ζ s - 1 / (s - 1)` tends to `γ` as `s → 1`. -/
+theorem _root_.tendsto_riemannZeta_sub_one_div :
+    Tendsto (fun s ↦ riemannZeta s - 1 / (s - 1)) (𝓝[≠] 1) (𝓝 eulerMascheroniConstant) := by
```

Comment:
```suggestion
    Tendsto (fun (s : ℂ) ↦ riemannZeta s - 1 / (s - 1)) (𝓝[≠] 1) (𝓝 eulerMascheroniConstant) := by
```
Lean doesn't need it, but it can help your reader to see the difference with the previous lemma.